infrahub_sdk/diff.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+)
+
+from typing_extensions import NotRequired, TypedDict
+
+
+class NodeDiff(TypedDict):
```

Comment:
FYI we have a:

```python
from infrahub_sdk.client import NodeDiff  # noqa: TCH002
```

In infrahub/message_bus/types.py, it would still work since we are importing NodeDiff into infrahub_sdk/client but it gets a bit weird.